src/crypto/tls/handshake_server_tls13.go

```diff
@@ -511,22 +506,14 @@ func (hs *serverHandshakeStateTLS13) pickCertificate() error {
 			// a Delegated Credential was not found. Fallback to the certificate.
 			return nil
 		}
-
 		if delegatedCredentialPair.DC != nil && delegatedCredentialPair.PrivateKey != nil {
 			// Even if the Delegated Credential has already been marshalled, be sure it is the correct one.
-			delegatedCredentialPair.DC.raw, err = delegatedCredentialPair.DC.marshal()
+			delegatedCredentialPair.DC.raw, err = delegatedCredentialPair.DC.Marshal()
 			if err != nil {
 				// invalid Delegated Credential. Fallback to the certificate.
 				return nil
 			}
-
-			hs.sigAlg, err = selectSignatureSchemeDC(c.vers, delegatedCredentialPair.DC, hs.clientHello.supportedSignatureAlgorithmsDC)
-			if err != nil {
-				// the Delegated Credential is unsupported or
-				// incompatible with the client's signature algorithms.
-				// Fallback to the certificate.
-				return nil
-			}
+			hs.sigAlg = delegatedCredentialPair.DC.cred.expCertVerfAlgo
```

Comment:
Nice. Note: `selectSignatureSchemeDC` checks that that support for `dc_expected_cert_verify_algorithm` was indicated by the client. I think it would be worth documenting this fact above `getDelegatedCredential()` so that it's clear to the reader.